gemrb/GUIScripts/pst/MessageWindow.py

```diff
@@ -31,76 +31,71 @@
-	GemRB.LoadWindowPack (GUICommon.GetWindowPack())
-	GemRB.SetInfoTextColor(0,255,0,255)
-	ActionsWindow = GemRB.LoadWindow(0)
-	OptionsWindow = GemRB.LoadWindow(2)
-	MessageWindow = GemRB.LoadWindow(7)
-	PortraitWindow = GUICommonWindows.OpenPortraitWindow (1)
-
-	MessageTA = MessageWindow.GetControl (1)
-	MessageTA.SetFlags (IE_GUI_TEXTAREA_AUTOSCROLL|IE_GUI_TEXTAREA_HISTORY)
-
-	GemRB.SetVar ("MessageTextArea", MessageTA.ID)
-	GemRB.SetVar ("ActionsWindow", ActionsWindow.ID)
-	GemRB.SetVar ("OptionsWindow", OptionsWindow.ID)
-	GemRB.SetVar ("MessageWindow", -1)
-	GemRB.SetVar ("OtherWindow", -1)
-	GemRB.SetVar ("ActionsPosition", 1) #Bottom
-	GemRB.SetVar ("OptionsPosition", 1) #Bottom
-	GemRB.SetVar ("MessagePosition", 1) #Bottom
-	GemRB.SetVar ("OtherPosition", 0) #Left
+	global MWindow, ActionsWindow, PortraitWindow, OptionsWindow
+
+	# TODO: we can uncomment the "HIDE_CUT" lines below to hide the windows for cutscenes
+	# the original doesn't hide them and it looks like there is a map drawing bug at the bottom of the screen due to the bottom
+	# row of tiles getting squished for not fitting perfectly on screen (tho I havent seen this in BG2, but maybe wasnt paying attention)
+
+	ActionsWindow = GemRB.LoadWindow(0, GUICommon.GetWindowPack(), WINDOW_BOTTOM|WINDOW_LEFT)
+	ActionsWindow.AddAlias("ACTWIN")
+	#ActionsWindow.AddAlias("HIDE_CUT", 1)
+	ActionsWindow.AddAlias("NOT_DLG", 0)
+	ActionsWindow.SetFlags(WF_BORDERLESS|IE_GUI_VIEW_IGNORE_EVENTS, OP_OR)
+
+	OptionsWindow = GemRB.LoadWindow(2, GUICommon.GetWindowPack(), WINDOW_BOTTOM|WINDOW_RIGHT)
+	OptionsWindow.AddAlias("OPTWIN")
+	#OptionsWindow.AddAlias("HIDE_CUT", 2)
+	OptionsWindow.AddAlias("NOT_DLG", 1)
+	OptionsWindow.SetFlags(WF_BORDERLESS|IE_GUI_VIEW_IGNORE_EVENTS, OP_OR)
 	GemRB.SetVar ("PortraitPosition", 1)
 
-	GemRB.GameSetScreenFlags (0, OP_SET)
-	
-	CloseButton= MessageWindow.GetControl (0)
+	MWindow = GemRB.LoadWindow(7, GUICommon.GetWindowPack(), WINDOW_BOTTOM|WINDOW_HCENTER)
+	MWindow.SetFlags(WF_DESTROY_ON_CLOSE, OP_NAND)
+	MWindow.AddAlias("MSGWIN")
+	MWindow.AddAlias("HIDE_CUT", 0)
+	MWindow.SetFlags(WF_BORDERLESS|IE_GUI_VIEW_IGNORE_EVENTS, OP_OR)
+
+	PortraitWindow = GUICommonWindows.OpenPortraitWindow (1, WINDOW_BOTTOM|WINDOW_HCENTER)
+	#PortraitWindow.AddAlias("HIDE_CUT", 3)
+	PortraitWindow.AddAlias("NOT_DLG", 2)
+	PortraitWindow.SetFlags(WF_BORDERLESS|IE_GUI_VIEW_IGNORE_EVENTS, OP_OR)
+
+	pframe = PortraitWindow.GetFrame()
+	pframe['x'] -= 16
+	PortraitWindow.SetFrame(pframe)
+
+	MessageTA = MWindow.GetControl (1)
+	MessageTA.SetFlags (IE_GUI_TEXTAREA_AUTOSCROLL|IE_GUI_TEXTAREA_HISTORY)
+	MessageTA.SetResizeFlags(IE_GUI_VIEW_RESIZE_ALL)
+	MessageTA.AddAlias("MsgSys", 0)
+
+	CloseButton= MWindow.GetControl (0)
 	CloseButton.SetText(28082)
-	CloseButton.SetEvent (IE_GUI_BUTTON_ON_PRESS, CommonWindow.OnDecreaseSize)
-	CloseButton.SetFlags (IE_GUI_BUTTON_DEFAULT, OP_OR)
+	CloseButton.SetEvent (IE_GUI_BUTTON_ON_PRESS, lambda: MWindow.Close())
```

Comment:
Codacy found an issue: [Lambda may not be necessary](https://app.codacy.com/manual/gemrb/gemrb/pullRequest?prid=6310305)